### PR TITLE
Let update_lints also generate the internal lints

### PR DESCRIPTION
**Pull request by flip1995**
r? @phansch 

changelog: none

### CODE REVIEW

File: clippy_dev/src/main.rs [hunk 1]
```diff
@@ -164,6 +164,8 @@ fn print_lints() {
 fn update_lints(update_mode: UpdateMode) {
     let lint_list: Vec<Lint> = gather_all().collect();
 
+    let internal_lints = Lint::internal_lints(lint_list.clone().into_iter());
+
     let usable_lints: Vec<Lint> = Lint::usable_lints(lint_list.clone().into_iter()).collect();
     let lint_count = usable_lints.len();
```
Comment (phansch) [inline clippy_dev/src/main.rs]: Might be worth renaming this to `usable_lint_count` to make clear it's only supposed to count usable lints.

**Comment by flip1995** [comment]
> I had concerns about internal lints showing up in user facing stuff.

With the explicit `.chain(internal_lints)` (like it is done somewhere with deprecated lints), this only affects the stuff in `clippy_lints/src/lib.rs`

**Comment by flip1995** [comment]
@bors treeclosed-

oops forgot to reopen the tree after doc update.

**Pull request merged**